Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -363,19 +757,151 @@ void SkylanderPortalWindow::UpdateEdits()
   }
 }
 
-bool SkylanderPortalWindow::eventFilter(QObject* object, QEvent* event)
+// helpers
+bool SkylanderPortalWindow::PassesFilter(QString name, u16 id, u16 var)
 {
-  // Close when escape is pressed
-  if (event->type() == QEvent::KeyPress)
+  IOS::HLE::USB::SkyData character = IOS::HLE::USB::list_skylanders.at(std::make_pair(id, var));
```

Comment:
```suggestion
 const auto character = IOS::HLE::USB::list_skylanders.find(std::make_pair(id, var));

  if (character == IOS::HLE::USB::list_skylanders.end())
  {
    return false;
  }
```

Just to ensure it isn't null, then in the if/else checks write character->second.game instead of character.game